### PR TITLE
Add LightInject support

### PR DESCRIPTION
**Pull request by jevgenigeurtsen**
Adds support for LightInject IoC container

https://github.com/seesharper/LightInject

### CODE REVIEW

**Comment by mausch** [comment]
Hi and thanks for the PR!
Can you squash this?

File: LightInject.SolrNet/LightInject.SolrNet.csproj [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <AssemblyName>LightInject.SolrNet</AssemblyName>
+    <RootNamespace>SolrNet</RootNamespace>
+    <PackageIconUrl>https://github.com/solrnet/solrnet/raw/master/Documentation/solr.png</PackageIconUrl>
+    <Authors>Gidon Junge and contributors</Authors>
```
Comment (mausch) [inline LightInject.SolrNet/LightInject.SolrNet.csproj:8]: You should put your name here :slightly_smiling_face:

**Comment by jevgenigeurtsen** [comment]
@mausch integration tests weren't running, so I've re-enabled running all integration tests (not just SolrNet ones).

Also added a new docker instance that enables auth so that authed request can be integration tested, I might add more tests for this.

**Comment by mausch** [comment]
Cherry-picked in c0b1f3c7f64d85eb702b0d143c9a2a327923554b , thanks

**Pull request closed without merging**